saw-central/src/SAWCentral/Builtins.hs

```diff
@@ -956,34 +965,43 @@ proveABC = do
   SV.AIGProxy proxy <- SV.scriptTopLevel SV.getProxy
   wrapProver [AIG] [] (Prover.proveABC proxy) Set.empty
 
-satExternal :: Bool -> String -> [String] -> ProofScript ()
+satExternal :: Bool -> Text -> [Text] -> ProofScript ()
 satExternal doCNF execName args =
   execTactic $ tacticSolve $ \g ->
     do SV.AIGProxy proxy <- SV.getProxy
        sc <- SV.getSharedContext
-       (mb, stats) <- Prover.abcSatExternal proxy sc doCNF execName args g
+       let execName' = Text.unpack execName
+           args' = map Text.unpack args
+       (mb, stats) <- Prover.abcSatExternal proxy sc doCNF execName' args' g
        case mb of
          Nothing -> return (stats, SolveSuccess (SolverEvidence stats (goalSequent g)))
          Just a  -> return (stats, SolveCounterexample a)
 
-writeAIGPrim :: FilePath -> Term -> TopLevel ()
-writeAIGPrim = Prover.writeAIG
+writeAIGPrim :: Text -> Term -> TopLevel ()
+writeAIGPrim ftxt e = do
+  let f :: FilePath = Text.unpack ftxt
+  Prover.writeAIG f e
 
-writeSAIGPrim :: FilePath -> TypedTerm -> TopLevel ()
-writeSAIGPrim file tt =
-  do write_tt <- Prover.writeSAIGInferLatches tt
-     io $ write_tt file
+writeSAIGPrim :: Text -> TypedTerm -> TopLevel ()
+writeSAIGPrim filetxt tt = do
+  let file :: FilePath = Text.unpack filetxt
+  write_tt <- Prover.writeSAIGInferLatches tt
+  io $ write_tt file
 
-writeSAIGComputedPrim :: FilePath -> Term -> Int -> TopLevel ()
-writeSAIGComputedPrim = Prover.writeSAIG
+writeSAIGComputedPrim :: Text -> Term -> Int -> TopLevel ()
+writeSAIGComputedPrim ftxt e n = do
+  let f :: FilePath = Text.unpack ftxt
+  Prover.writeSAIG f e n
 
 -- | Bit-blast a proposition check its validity using the RME library.
 proveRME :: ProofScript ()
 proveRME = wrapProver [RME] [] Prover.proveRME Set.empty
 
-codegenSBV :: SharedContext -> FilePath -> [String] -> String -> TypedTerm -> TopLevel ()
-codegenSBV sc path unints fname (TypedTerm _schema t) =
-  do unintSet <- resolveNames unints
+codegenSBV :: SharedContext -> Text -> [Text] -> Text -> TypedTerm -> TopLevel ()
+codegenSBV sc pathtxt unints fnametxt (TypedTerm _schema t) = do
+     let path :: FilePath = Text.unpack pathtxt
+         fname :: FilePath = Text.unpack fnametxt
+     unintSet <- resolveNames (map Text.unpack unints)
```

Comment:
`resolveNames` and `resolveName` feel like prime opportunities for making their arguments take `Text` instead of `String`, as that would avoid a _lot_ of intermediate `Text.unpack` calls.